lib/internal/fs/streams.js

```diff
@@ -103,7 +104,7 @@ function ReadStream(path, options) {
   }
 
   if (typeof this.fd !== 'number')
-    this.open();
+    openReadFs(this);
```

Comment:
graceful-fs provides replacements for `ReadStream` and `WriteStream` with overridden `open` methods to retry on `ENFILE` / `EMFILE` errors, this change would make that ineffective.  The only exception would be if graceful-fs were told to directly patch `require('fs')`.

For future versions in theory graceful-fs would have to deal with this by reimplementing the ReadStream / WriteStream constructors to use the graceful-fs open method.  I say this assuming node.js is not open to accepting an `fs` option on the constructors to allow using an alternative fs implementation, in that case graceful-fs could inject the option before calling the core constructor.

Reply:
https://github.com/nodejs/node/pull/29083

Reply:
Updated PR to not break graceful-fs